.github/workflows/add_milestone.yml

```diff
@@ -0,0 +1,49 @@
+name: Add Milestone on a Merged PR
+
+on:
+  push:
+    branches:
+      - main
+      - "[0-9]+.[0-9]+.x"
+
+jobs:
+  add-milestone-pr:
+    name: Add Milestone on PR
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get PR from which the commit is from
+        id: source-pr
+        run: echo "NUMBER=$(gh pr list --search "$GITHUB_SHA" --state merged --json url --jq '.[0].url' | sed 's|.*pull/\(.*\)|\1|g')" >> "$GITHUB_OUTPUT"
+      - name: Get repo current milestone
+        id: current-milestone
+        run: |
+          if [[ ${GITHUB_REF##*/} =~ ^7\.[0-9]+\.[0-9]+$ ]]; then
+            # If we're on a release branch, set the milestone to the latest release milestone found.
+            MILESTONE=$(gh release list | grep -o $(echo ${GITHUB_REF##*/} | sed 's/x/[0-9]*/g') | sort -uV | tail -1)
+          else
+            # Else use the current_milestone field in the release.json file.
+            MILESTONE=$(cat release.json | jq -r .current_milestone)" >> "$GITHUB_OUTPUT"
+          fi
+          echo "MILESTONE=$MILESTONE" >> "$GITHUB_OUTPUT"
+
+      - name: Set the merged PR milestone to current_milestone from release.json
+        run: |
+          if [ -z "$NUMBER" ]; then
+            echo "Error: Couldn't find the PR commit $GITHUB_SHA originating from, milestone not assigned."
+            exit 1
+          elif [ -z "$MILESTONE" ]; then
+            echo "Error: Couldn't find the current_milestone field in the release.json file."
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
What about logging the error at the previous step (line 22 and 25), so that we know if it failed reading the `release.json` file or the `release list` ?

Reply:
Nice catch, thanks ! Made it cleaner in [b2ced26](https://github.com/DataDog/datadog-agent/pull/22567/commits/b2ced268c4e8400056e5a050a0e979b233aaf561) !